src/NuGet.DependencyResolver/SourceRepositoryDependencyProvider.cs

```diff
@@ -76,7 +76,11 @@ public class SourceRepositoryDependencyProvider : IRemoteDependencyProvider
 
             using (var nupkgStream = await _findPackagesByIdResource.GetNupkgStreamAsync(identity.Name, identity.Version, cancellationToken))
             {
-                await nupkgStream.CopyToAsync(stream, bufferSize: 8192, cancellationToken: cancellationToken);
+                cancellationToken.ThrowIfCancellationRequested();
+
+                // If the stream is already available, do not stop in the middle of copying the stream
```

Comment:
This is the last point at which we can cancel the restore safely. Do you ask this, because, you consider that package download to be the most expensive part of the restore operation?

If you are asking about passing CancellationToken.None to Stream.CopyToAsync,
I don't want to pass in the real CancellationToken to Stream.CopyToAsync. I think it will stop the copy abruptly if the CancellationToken was cancelled. Previously, this was never an issue, because, well, we were not passing the real CancellationToken